seamm_datastore/api.py

```diff
@@ -421,31 +421,44 @@ def get_groups(_=None, as_json=False, current_user=None):
     return groups
 
 
-def get_jobs(_=None, as_json=False, current_user=None, limit=None):
+def get_jobs(_=None, as_json=False, limit=None, offset=None, count=False, status=None):
+
     from seamm_datastore.database.models import Job, Project
 
-    jobs = Job.query.all()
-
-    # After we get the jobs, we need to know which jobs belong to projects which the
-    # user can read. This might be a performance issue on a larger DB, but I think we
-    # can do this check here.
-    authorized_jobs = Job.query.filter(Job.authorized("read")).all()
-    auth_projects = Project.query.filter(Project.authorized("read")).all()
-    for job in jobs:
-        if job not in authorized_jobs:
-            for project in job.projects:
-                if project in auth_projects:
-                    authorized_jobs.append(job)
-                    # We can exit the loop if we have found one
-                    # project which grants read permission
-                    break
+    # Get the jobs where the user has read permission
+    if status is None:
+        authorized_jobs = Job.query.filter(Job.authorized("read"), Job.status == status)
+    else:
+        authorized_jobs = Job.query.filter(Job.authorized("read"), Job.status == status)
 
-    if as_json:
-        from seamm_datastore.database.schema import JobSchema
+    # Get jobs from projects where user has read permission
+    if status is None:
+        project_jobs = Project.query.with_entities(Job).filter(
+            Project.authorized("read")
+        )
+    else:
+        project_jobs = Project.query.with_entities(Job).filter(
+            Project.authorized("read"), Job.status == status
+        )
+
+    # Find union of these queries
+    jobs = authorized_jobs.union(project_jobs)
+
+    # Continue building
+    if limit is not None:
+        jobs = jobs.limit(limit)
+    if offset is not None:
+        jobs = jobs.offset(offset)
+
```

Comment:
We had mentioned separating `count` from `get`. If you took all of the method up to this point and called it e.g. `_job_query`, returning the `jobs` object then `count_jobs` could be a small wrapper calling `jobs = _job_query(...)` and then `return jobs.count` while `get_jobs` could be a small method with the call to `_job_query` and lines 456+ below.

If we wanted another variant, e.g. `list_job_names` it would be just another little wrapper.... This approach would work with all the methods (project, user, ...) as far as I can see.

Reply:
Yes, I agree with separating `count` from `get`. I agree with the approach of an inner "private" function (beginning with `_`). For now, I was only updating this one function and making it consistent with the other function. 

Options are:
1. Add `count` functions in this PR.
2. Merge PR with `count` in `get` functions and separate out in next PR.

I'm kind of leaning towards number 2 above. Would you like it in this PR?

Reply:
Whatever is convenient for you! I don't use count (yet) so nothing will break for me -- and if you make a change to the API it won't be hard to fix the code in the SEAMM GUI. 

A functionality that I would like to get to fairly soon in the GUI is to allow people to open flowcharts directly from the Dashboard. This will need a `get_flowchart` API call that returns the flowchart in some reasonable form (?string? -- I don't think it can be JSON directly because it is a bit more complicated) While we do have a unique list of flowcharts, I think the approach that makes most sense from the user's point of view would be to provide an optionally filtered list of job numbers and titles and perhaps the date/time it was run, let them choose from that list, or directly enter a job number if they are looking through the web interface, and get the Flowchart for that job.

I think that we have the `get_flowchart` api, but not quite the `list_jobs` api -- though maybe using the JSON from the current `get_jobs` would be an OK way to go. I guess the JSON is not huge, so handling on the order of 1,000 jobs would be no problem (and beyond that we'd need to page through a list -- inferring 100k jobs is not useful).

So handle the current changes as you see fit, and also feel free to change the API if needed. Just let me know, and I will fix the GUI to work with the changed API before making a new release.